aws/resource_aws_cognito_identity_provider.go

```diff
@@ -0,0 +1,180 @@
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"provider_type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"user_pool_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAwsCognitoIdentityProviderCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cognitoidpconn
+	log.Print("[DEBUG] Creating Cognito Identity Provider")
+
+	name := aws.String(d.Get("provider_name").(string))
+	params := &cognitoidentityprovider.CreateIdentityProviderInput{
+		ProviderName: name,
+		ProviderType: aws.String(d.Get("provider_type").(string)),
+		UserPoolId:   aws.String(d.Get("user_pool_id").(string)),
+	}
+
+	if v, ok := d.GetOk("attribute_mapping"); ok {
+		params.AttributeMapping = expandCognitoIdentityProviderMap(v.(map[string]interface{}))
+	}
+
+	if v, ok := d.GetOk("provider_details"); ok {
+		params.ProviderDetails = expandCognitoIdentityProviderMap(v.(map[string]interface{}))
+	}
+
+	if v, ok := d.GetOk("idp_identifiers"); ok {
+		params.IdpIdentifiers = expandStringList(v.([]interface{}))
+	}
+
+	_, err := conn.CreateIdentityProvider(params)
+	if err != nil {
+		return fmt.Errorf("Error creating Cognito Identity Provider: %s", err)
+	}
+
+	d.SetId(*name)
+
+	return resourceAwsCognitoIdentityProviderRead(d, meta)
+}
+
+func resourceAwsCognitoIdentityProviderRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cognitoidpconn
+	log.Printf("[DEBUG] Reading Cognito Identity Provider: %s", d.Id())
+
+	ret, err := conn.DescribeIdentityProvider(&cognitoidentityprovider.DescribeIdentityProviderInput{
+		ProviderName: aws.String(d.Id()),
+		UserPoolId:   aws.String(d.Get("user_pool_id").(string)),
+	})
+
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "ResourceNotFoundException" {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	ip := ret.IdentityProvider
+	d.Set("provider_name", ip.ProviderName)
+	d.Set("provider_type", ip.ProviderType)
+	d.Set("user_pool_id", ip.UserPoolId)
+
+	if err := d.Set("attribute_mapping", flattenCognitoIdentityProviderMap(ip.AttributeMapping)); err != nil {
+		return fmt.Errorf("[DEBUG] Error setting attribute_mapping error: %#v", err)
+	}
+
+	if err := d.Set("provider_details", flattenCognitoIdentityProviderMap(ip.ProviderDetails)); err != nil {
+		return fmt.Errorf("[DEBUG] Error setting provider_details error: %#v", err)
+	}
+
+	if err := d.Set("idp_identifiers", flattenStringList(ip.IdpIdentifiers)); err != nil {
+		return fmt.Errorf("[DEBUG] Error setting idp_identifiers error: %#v", err)
+	}
+
+	return nil
+}
+
+func resourceAwsCognitoIdentityProviderUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cognitoidpconn
+	log.Print("[DEBUG] Updating Cognito Identity Provider")
+
+	params := &cognitoidentityprovider.UpdateIdentityProviderInput{
+		UserPoolId:   aws.String(d.Get("UserPoolId").(string)),
```

Comment:
Just wondering... is "d.Get("UserPoolId") correct? Or should it be: d.Get("user_pool_id")?

Reply:
Yes, I will change it.